packages/react-reconciler/src/__tests__/useSyncExternalStore-test.js

```diff
@@ -292,4 +353,88 @@ describe('useSyncExternalStore', () => {
       );
     },
   );
+
+  it('regression: doesnt loop for only changing store reference', async () => {
+    let store = {validationResults: new Map(), version: 0};
+    let listeners = [];
+
+    const ExternalStore = {
+      set(value) {
```

Comment:
is it intentional to not use `value`?

Reply:
Yeah, we could set it with `setTimeout`, which is basically what some stores do but with updating the value in effects.